internal/pkg/api/cloudstack.go

```diff
@@ -124,6 +124,19 @@ func WithCloudStackAccount(value string) CloudStackFiller {
 	}
 }
 
+// WithCloudStackCredentialsRef returns a CloudStackFiller that updates the edentialsRef of all availability zones.
+func WithCloudStackCredentialsRef(value string) CloudStackFiller {
+	return func(config CloudStackConfig) {
+		zones := []anywherev1.CloudStackAvailabilityZone{}
+		for _, az := range config.datacenterConfig.Spec.AvailabilityZones {
+			az.CredentialsRef = value
+			zones = append(zones, az)
+		}
+
+		config.datacenterConfig.Spec.AvailabilityZones = zones
```

Comment:
Updating the availability zones from in the loop wasn't enough. The AvailabilityZones value needed to be fully replace for the updates to be reflected.